src/lib.rs

```diff
@@ -289,6 +289,62 @@ impl<'a> DoubleArrayTrieBuilder<'a> {
     }
 }
 
+pub struct PrefixIter<'a> {
+    key_len: usize,
+    da: &'a DoubleArrayTrie,
+    char_indices: str::CharIndices<'a>,
+    b: i32,
+    n: i32,
+    p: usize,
+    reach_leaf: bool,
+}
+
+impl<'a> Iterator for PrefixIter<'a> {
+    type Item = (usize, usize);
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
``.size_hint()`` can be implemented with ``longest_word_len`` to speed up ``Iterator::collect::<Vec<_>>()``